src/compiler/crystal/semantic/call_error.cr

```diff
@@ -37,6 +37,9 @@ class Crystal::Path
 end
 
 class Crystal::Call
+  # :nodoc:
+  MAX_RENDERED_OVERLOADS = 500
```

Comment:
Most terminals have can scroll up 500 lines, this means it won't be possible to read the full error message.

I would suggest 10 or 20.